File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    
    arr = [1]*(n+1)
    
    lost.sort()

    for i in range(1, n+1):
        if i in lost and i in reserve:
            arr[i] = 1
        elif i in lost:
            arr[i] = 0
        elif i in reserve:
            arr[i] = 2
    
    for k in lost:
        if k > 1 and arr[k-1] == 2:
            arr[k-1] = 1
            arr[k] = 1
        elif k < n and arr[k+1] == 2:
            arr[k+1] = 1
            arr[k] = 1
    lo = arr.count(0)

    return n-lo

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;
class Solution {
    public static int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        int[] people = new int[n+2];
        people[0] = -1;

        for(int l : lost)
            people[l] -= 1;
        for(int r : reserve)
            people[r] += 1;

        for(int i=1; i<n+1; i++){
            if(people[i]>=0){
                continue;
            }else if(people[i] == -1 && people[i-1] >0){
                people[i] += 1;
                people[i-1] -= 1;
            }else if(people[i] == -1 && people[i+1] >0){
                people[i] += 1;
                people[i+1] -= 1;
            }else{
                answer -= 1;
            }
        }

        return answer;
    }
}